mxcubecore/HardwareObjects/TangoShutter.py

```diff
@@ -114,36 +127,43 @@ def _initialise_values(self):
                 "FAULT": "FAULT",
             }
         )
+        try:
+            config_values = json.loads(self.get_property("values"))
+            for key, val in config_values.items():
+                if isinstance(val, (tuple, list)):
+                    values_dict.update({key: val[1]})
+                    states_dict.update({key: (HardwareObjectState[val[1]], val[0])})
+                else:
+                    values_dict.update({key: val})
+        except (ValueError, TypeError):
+            logging.exception("Exception in _initialise_values()")
+
         self.VALUES = Enum("ValueEnum", values_dict)
+        self.SPECIFIC_STATES = Enum("TangoShutterStates", states_dict)
 
-    def get_state(self):
+    def get_state(self) -> HardwareObjectState:
         """Get the device state.
+
         Returns:
-            (enum 'HardwareObjectState'): Device state.
+            Device state.
         """
         try:
-            if self.config_values:
-                _state = self.config_values[str(self.state_channel.get_value())]
-            else:
-                _state = str(self.state_channel.get_value())
-
+            _state = self.get_value().name
+            return self.SPECIFIC_STATES[_state].value[0]
         except (AttributeError, KeyError):
+            logging.exception("Exception in get_state()")
```

Comment:
This is strange to me.
Feels like it should have triggered Ruff rule `LOG015`: <https://docs.astral.sh/ruff/rules/root-logger-call/>.

```suggestion
            hwr_logger.exception("Exception in get_state()")
```

With somewhere towards the top, after the imports, something like this:

```
hwr_logger = logging.getLogger("HWR")
```

Reply:
> Feels like it should have triggered Ruff rule `LOG015`: https://docs.astral.sh/ruff/rules/root-logger-call/.

Ah okay, it says right there that it is a "preview" rule that needs the `--preview` flag to be enabled.

Still should be fixed, though.

Reply:
Well spotted :D Fixed